modules/tensor_mechanics/src/materials/RecomputeRadialReturnIsotropicPowerLawHardening.C

```diff
@@ -0,0 +1,80 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "RecomputeRadialReturnIsotropicPowerLawHardening.h"
+
+template<>
+InputParameters validParams<RecomputeRadialReturnIsotropicPowerLawHardening>()
+{
+  InputParameters params = validParams<RecomputeRadialReturnIsotropicPlasticity>();
+
+  // Set and Suppress paramaters to enable calculation of the yield stress
+  params.set<Real>("yield_stress") = 1.0;
+  params.set<Real>("hardening_constant") = 1.0;
+  params.suppressParameter<Real>("yield_stress");
+  params.suppressParameter<Real>("hardening_constant");
+
+  // Power law hardening specific parameters
+  params.addRequiredParam<Real>("strength_coefficient", "The strength coefficient (K) for power law hardening");
+  params.addRequiredParam<Real>("strain_hardening_exponent", "The strain hardening exponent (n) for power law hardening");
+
+  return params;
+}
+
+
+RecomputeRadialReturnIsotropicPowerLawHardening::RecomputeRadialReturnIsotropicPowerLawHardening(const InputParameters & parameters) :
+    RecomputeRadialReturnIsotropicPlasticity(parameters),
+    _K(parameters.get<Real>("strength_coefficient")),
+    _strain_hardening_exponent(parameters.get<Real>("strain_hardening_exponent"))
+{
+  if (_strain_hardening_exponent <= 0.0  || _strain_hardening_exponent >= 1.0)
+    mooseError("Verify the value of the strain hardening exponent, which should be a value between 0.0 and 1.0");
+}
+
+void
+RecomputeRadialReturnIsotropicPowerLawHardening::computeStressInitialize(Real effectiveTrialStress)
+{
+  _shear_modulus = getIsotropicShearModulus();
+  computeYieldStress();
+
+  _effective_trial_stress = effectiveTrialStress;
+  _yield_condition = effectiveTrialStress - _hardening_variable_old[_qp] - _yield_stress;
+  _hardening_variable[_qp] = _hardening_variable_old[_qp];
+  _plastic_strain[_qp] = _plastic_strain_old[_qp];
+}
+
+Real
+RecomputeRadialReturnIsotropicPowerLawHardening::computeHardeningDerivative(Real scalar)
+{
+  Real stress_delta = _effective_trial_stress - 3.0 * _shear_modulus * scalar;
+  Real slope = std::pow(stress_delta, (1.0 / _strain_hardening_exponent - 1.0)) / _strain_hardening_exponent * 1.0 / std::pow(_K, 1.0 / _strain_hardening_exponent);
+  slope -= 1.0 / _youngs_modulus;
+
+  return 1.0 / slope;
+}
+
+void
+RecomputeRadialReturnIsotropicPowerLawHardening::computeYieldStress()
+{
+  // Pull in the Lame lambda, and caculate E
```

Comment:
Should we try making this `Lamé`. Our minimum compiler requirements should be able to handle this.